src/topp/PhosphoScoring.cpp

```diff
@@ -93,15 +92,150 @@ class TOPPPhosphoScoring :
 
 protected:
 
+  // spectrum must not contain 0 intensity peaks and must be sorted by m/z
+  template <typename SpectrumType>
+  static void deisotopeAndSingleChargeMSSpectrum_(SpectrumType& in, Int min_charge, Int max_charge, double fragment_tolerance, bool fragment_unit_ppm, bool keep_only_deisotoped = false, Size min_isopeaks = 3, Size max_isopeaks = 10, bool make_single_charged = true)
```

Comment:
I hope that we can (now or in the future) get a real de-isotoping functionality into OpenMS for single spectra. This is fine for now but I think this is currently missing in the library and would be nice to transfer this into its own class at some point. what do you think?